### PR TITLE
Fix possible importing errors in build_libtorch.py

### PR DESCRIPTION
**Pull request by bombs-kim**
1. I fixed the importing process, which had some problems
    -  **I think `setup_helpers` should not be imported as the top level module. It can lead to many future errors. For example, what if `setup_helpers` imports another module from the upper level?** So we need to change it.
    - The code is not consistent with other modules in `tools` package. For example, other
    modules in the package imports `from tools.setuptools...` not `from setuptools...`.
    - **It should be able to run with `python -m tools.build_libtorch` command**  because this module is a part of the tools package. Currently, you cannot do that and I think it's simply wrong.

~~2. I Added platform specific warning messages.
    - I constantly forgot that I needed to define some environment variables in advance specific to my platform to build libtorch, especially when I'm working at a non pytorch root directory. So I thought adding warnings for common options would be helpful .~~

~~3. Made the build output path configurable. And a few other changes.~~

@orionr  @ebetica

### CODE REVIEW

**Comment by orionr** [comment]
Thank you for this. cc @goldsborough as well. Are the CI failures related?

**Comment by bombs-kim** [comment]
@orionr  I'm not sure if the CI failures related with this PR. I think I have to revert some changes and see if there is still an error. Or is there easier way to run those CI test on my environment?

**Comment by bombs-kim** [comment]
I reverted some changes but still it couldn't pass some tests.

<details>
<summary>17 failing and 41 successful checks

ci/circleci: caffe2_py2_cuda8_0_cudnn6_ubuntu16_04_build — Your tests failed on CircleCI
Details ... (click)</summary>

ci/circleci: pytorch_linux_trusty_py2_7_9_test — Your tests failed on CircleCI
Details

ci/circleci: pytorch_linux_trusty_py2_7_test — Your tests failed on CircleCI
Details

ci/circleci: pytorch_linux_trusty_py3_5_test — Your tests failed on CircleCI
Details

ci/circleci: pytorch_linux_trusty_py3_6_gcc4_8_test — Your tests failed on CircleCI
Details

ci/circleci: pytorch_linux_trusty_py3_6_gcc5_4_test — Your tests failed on CircleCI
Details

ci/circleci: pytorch_linux_trusty_py3_6_gcc7_build — Your tests failed on CircleCI
Details

ci/circleci: pytorch_linux_trusty_pynightly_test — Your tests failed on CircleCI
Details

ci/circleci: pytorch_linux_xenial_cuda8_cudnn6_py3_NO_AVX2_test — Your tests failed on CircleCI
Details

ci/circleci: pytorch_linux_xenial_cuda8_cudnn6_py3_NO_AVX_NO_AVX2_test — Your tests failed on CircleCI
Details

ci/circleci: pytorch_linux_xenial_cuda8_cudnn6_py3_test — Your tests failed on CircleCI
Details

ci/circleci: pytorch_linux_xenial_cuda9_2_cudnn7_py3_gcc7_test — Your tests failed on CircleCI
Details

ci/circleci: pytorch_linux_xenial_cuda9_cudnn7_py2_test — Your tests failed on CircleCI
Details

ci/circleci: pytorch_linux_xenial_cuda9_cudnn7_py3_test — Your tests failed on CircleCI
Details

ci/circleci: pytorch_linux_xenial_py3_clang5_asan_test — Your tests failed on CircleCI
Details

ci/circleci: pytorch_macos_10_13_py3_test — Your tests failed on CircleCI
Details
@pytorchbot
pr/pytorch-win-ws2016-cuda9-cudnn7-py3 — Build failed
Details

ci/circleci: caffe2_onnx_py2_gcc5_ubuntu16_04_build — Your tests passed on CircleCI!
Details

ci/circleci: caffe2_onnx_py2_gcc5_ubuntu16_04_test — Your tests passed on CircleCI!
Details

ci/circleci: caffe2_py2_android_ubuntu16_04_build — Your tests passed on CircleCI!
Details

ci/circleci: caffe2_py2_clang3_8_ubuntu16_04_build — Your tests passed on CircleCI!
Details

ci/circleci: caffe2_py2_clang3_9_ubuntu16_04_build — Your tests passed on CircleCI!
Details

ci/circleci: caffe2_py2_clang7_ubuntu16_04_build — Your tests passed on CircleCI!
Details

ci/circleci: caffe2_py2_cuda8_0_cudnn7_ubuntu16_04_build — Your tests passed on CircleCI!
Details

ci/circleci: caffe2_py2_cuda9_0_cudnn7_centos7_build — Your tests passed on CircleCI!
Details

ci/circleci: caffe2_py2_cuda9_0_cudnn7_centos7_test — Your tests passed on CircleCI!
Details

ci/circleci: caffe2_py2_cuda9_0_cudnn7_ubuntu16_04_build — Your tests passed on CircleCI!
Details

ci/circleci: caffe2_py2_cuda9_0_cudnn7_ubuntu16_04_test — Your tests passed on CircleCI!
Details

ci/circleci: caffe2_py2_cuda9_1_cudnn7_ubuntu16_04_build — Your tests passed on CircleCI!
Details

ci/circleci: caffe2_py2_cuda9_1_cudnn7_ubuntu16_04_test — Your tests passed on CircleCI!
Details

ci/circleci: caffe2_py2_gcc4_8_ubuntu14_04_build — Your tests passed on CircleCI!
Details

ci/circleci: caffe2_py2_gcc4_8_ubuntu14_04_test — Your tests passed on CircleCI!
Details

ci/circleci: caffe2_py2_ios_macos10_13_build — Your tests passed on CircleCI!
Details

ci/circleci: caffe2_py2_mkl_ubuntu16_04_build — Your tests passed on CircleCI!
Details

ci/circleci: caffe2_py2_mkl_ubuntu16_04_test — Your tests passed on CircleCI!
Details

ci/circleci: caffe2_py2_system_macos10_13_build — Your tests passed on CircleCI!
Details

ci/circleci: pytorch_doc_push — Your tests passed on CircleCI!
Details

ci/circleci: pytorch_linux_trusty_py2_7_9_build — Your tests passed on CircleCI!
Details

ci/circleci: pytorch_linux_trusty_py2_7_build — Your tests passed on CircleCI!
Details

ci/circleci: pytorch_linux_trusty_py3_5_build — Your tests passed on CircleCI!
Details

ci/circleci: pytorch_linux_trusty_py3_6_gcc4_8_build — Your tests passed on CircleCI!
Details

ci/circleci: pytorch_linux_trusty_py3_6_gcc5_4_build — Your tests passed on CircleCI!
Details

ci/circleci: pytorch_linux_trusty_pynightly_build — Your tests passed on CircleCI!
Details

ci/circleci: pytorch_linux_xenial_cuda10_cudnn7_py3_gcc7_build — Your tests passed on CircleCI!
Details

ci/circleci: pytorch_linux_xenial_cuda8_cudnn6_py3_build — Your tests passed on CircleCI!
Details

ci/circleci: pytorch_linux_xenial_cuda8_cudnn6_py3_multigpu_test — Your tests passed on CircleCI!
Details

ci/circleci: pytorch_linux_xenial_cuda9_2_cudnn7_py3_gcc7_build — Your tests passed on CircleCI!
Details

ci/circleci: pytorch_linux_xenial_cuda9_cudnn7_py2_build — Your tests passed on CircleCI!
Details

ci/circleci: pytorch_linux_xenial_cuda9_cudnn7_py3_build — Your tests passed on CircleCI!
Details

ci/circleci: pytorch_linux_xenial_py3_clang5_asan_build — Your tests passed on CircleCI!
Details

ci/circleci: pytorch_macos_10_13_cuda9_2_cudnn7_py3_build — Your tests passed on CircleCI!
Details

ci/circleci: pytorch_macos_10_13_py3_build — Your tests passed on CircleCI!
Details

ci/circleci: pytorch_short_perf_test_gpu — Your tests passed on CircleCI!
Details

continuous-integration/travis-ci/pr — The Travis CI build passed
Details
@pytorchbot
pr/caffe2-py2-clang7-rocmdeb-ubuntu16.04-test — Build successful
Details
@pytorchbot
pr/caffe2-py2-cuda9.0-cudnn7-windows-build — Build successful
Details
@pytorchbot
pr/caffe2-py2-devtoolset7-rocmrpm-centos7.5-test — Build successful
Details
@pytorchbot
pr/py2-clang7-rocmdeb-ubuntu16.04 — Build successful
Details
</details>

Now I'm reverting all code changes, and try the tests again.

**Comment by bombs-kim** [comment]
@orionr
The set of tests changed while I updated the reverted commits. The overall result is almost identical though. I think those CI failures may not be related with this PR. It's maybe because I branched out from a quite old commit of the master branch, I guess?

(Changes in test results)
image

**Comment by ezyang** [comment]
@bombs-kim our CI doesn't automatically rebase/merge for you. So try merging with master and pushing again.

**Comment by bombs-kim** [comment]
@ezyang Thank you, I merged the current branch with the master and pushed it.

**Comment by bombs-kim** [comment]
@goldsborough
@ezyang 
I've updated the code to keep up with the latest master branch🙂

File: docs/libtorch.rst [hunk 1]
```diff
@@ -1,17 +1,23 @@
 libtorch (C++-only)
 ===================
 
-The core of pytorch can be built and used without Python. A
```
Comment (goldsborough) [inline docs/libtorch.rst]: Unnecessary "that" in"that does not depend on"

**Review by ezyang — APPROVED** [review]
okey dokey

**Pull request closed without merging**